service/src/main_impl.rs

```diff
@@ -104,6 +104,13 @@ pub(crate) fn main() {
 
 	info!("*** Running worker in mode: {:?} \n", WorkerModeProvider::worker_mode());
 
+	let mut lockfile = PathBuf::from(config.data_dir());
+	lockfile.push("worker.lock");
+	while std::fs::metadata(lockfile.clone()).is_ok() {
+		println!("lockfile is present, will wait for it to disappear {:?}", lockfile);
+		thread::sleep(std::time::Duration::from_secs(5));
+	}
```

Comment:
k8 flow:
1. create `worker.lock` file in workdir
2. restart service. will wait for lock to disappear
3. do maintenance tasks (snapshotting, db restore, whatever)
4. delete `worker.lock` file 
5. worker will startup after <5s

Reply:
Shouldn't this be documented in the code instead of a review comment?

Reply:
yeah, well. rather devops docs in confluence